build.rs

```diff
@@ -438,7 +438,10 @@ fn build_unix(target_info: &TargetInfo, out_dir: PathBuf) {
                   out_dir.clone(),
                   test_header_change);
 
-    let libcxx = if target_info.target_vendor() == "apple" ||
+    let libcxx = if target_info.target_vendor()
+        .map(|v| v == "apple")
```

Comment:
I'd guess that the target vendor doesn't matter if we can detect target_os == "macos" or "ios". Conversely, I think that if we do this for "apple" then I think we wouldn't need to care what the target OS is. So either this logic is too complex or we need to have a comment about it as it is non-obvious.

Also, do you have a mac? If you don't, then don't worry about the Mac failures. I will fix the Mac build for you. Just concentrate on Linux.